### DISCUSSION TITLE
what would it take to get rid of Dialect.getSqlTypeDescriptorOverride() ?

### DISCUSSION THREAD

**Discussion by gavinking** [hibernate/hibernate-orm#3847] [Design Proposals]
I really hate `Dialect.getSqlTypeDescriptorOverride()`, and I highly suspect that at least _some_ of the overrides that exist in the `Dialect`s either aren't necessary or could be done in a different way. 

Indeed, as I understand it, a lot of those overrides are there *only* to make schema update/validation work when the db doesn't distinguish natively between two JDBC types. And it seems to me that replacing the whole type descriptor is just not the best way to solve that problem.

At the very least, can't this be handled via `contributeTypes()` instead? Why do we have two ways of doing what looks like much the same thing?

**Comment by sebersole** [comment]
I totally I agree.  The remapping API stuff was just the way I could backport these remapping ideas from 6 (really even the "pre work" for 6) to fit within restrictions of compatibility.

But that stuff could and should just go away.

**Reply by beikov** [reply]
It seems to me, that what I have done for Oracle (override for BOOLEAN in `Dialect#contributeTypes`) could be used in all the other Dialects as well for their overrides, just like Gavin suggests. If we all agree, I can create a PR for this.

**Reply by gavinking** [reply]
I think that would be a good start, but on the other hand I don't want to just swap type overrides for type contributions. I think we should also review what's there and if they're really necessary. It's a little tricky because the overrides that are there today don't do a good job of documenting what their purposes are.

**Reply by sebersole** [reply]
The contributions are actually exactly the correct replacement for the legacy remapping.

I can't say if the Dialects themselves do a good job documenting why they register/contribute certain overrides, but they absolutely should

**Reply by gavinking** [reply]
What I mean is if for example one of them is not necessary at all, or only there to e.g. make schema validator happy, I would prefer to remove it / replace it with a different solution.

**Reply by sebersole** [reply]
considering the dialect also defines its exporter for schema export, i think that's reasonable

**Reply by sebersole** [reply]
come to think of it... we should probably have dialect provide similar for how to do validation and update as well for the some of the same reasons
